client/layout/index.jsx

```diff
@@ -391,7 +391,9 @@ export default compose(
 		const shouldShowAppBanner = getShouldShowAppBanner( getSelectedSite( state ) );
 		const sectionJitmPath = getMessagePathForJITM( currentRoute );
 		const isJetpackLogin = startsWith( currentRoute, '/log-in/jetpack' );
-		const isJetpack = isJetpackSite( state, siteId ) && ! isAtomicSite( state, siteId );
+		const isJetpack =
+			( isJetpackSite( state, siteId ) && ! isAtomicSite( state, siteId ) ) ||
+			startsWith( currentRoute, '/checkout/jetpack' );
```

Comment:
The checkout component rendered when a user visits `/checkout/jetpack` doesn't have a selected site in context, so it doesn't know it's dealing with a Jetpack site. The result is that the masterbar is blue, not green.

The component doesn't have a site in context because it is connected at the site level, which means it won't be available on WP.com until a subscription is purchased.